assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+        }
+    }
+
+    public static void gettingTopPerformers() {
+        ArrayList<Player> top3Batsmen = getTopPlayersByRole("BATSMEN");
+        ArrayList<Player> top3Bowler = getTopPlayersByRole("BOWLER");
+        ArrayList<Player> top3AllRounder = getTopPlayersByRole("ALL ROUNDER");
+        displayArrayList(top3Batsmen);
+    }
+
+    static void displayArrayList(ArrayList<Player> players) {
+        for (Player player : players) {
+            player.display();
+        }
+    }
+
+    public static ArrayList<Player> getTopPlayersByRole(String role) {
+        Stack<Player> top3 = new Stack<>();
+        ArrayList<Player> result = new ArrayList<>();
+        for (Team team : teams) {
+            for (Player p : team.squad) {
+                if (p.role.equals(role)) {
+                    if (top3.size() == 0) {
+                        top3.push(p);
+                    } else {
+                        Stack<Player> temp = new Stack<>();
+                        while (!top3.isEmpty() && top3.peek().runs < p.runs) {
+                            temp.push(top3.pop());
+                        }
+                        temp.push(p);
+                        while (!temp.isEmpty()) {
+                            top3.push(temp.pop());
+                        }
+                        while (top3.size() > 3) {
+                            top3.pop();
+                        }
+                    }
+                }
+            }
+        }
+        while (!top3.isEmpty()) {
+            result.add(top3.pop());
+        }
+        return result;
+    }
+
+    public static void bestPossiblePerformance() {
+        HashMap<Team, Double> mapTeamToPredictedScore = new HashMap<>();
+        for (Team team : teams) {
+            Comparator<Player> cmp = Comparator.comparing(Player::getRuns);
+            team.squad.sort(cmp.reversed());
+            Double predictedScore = 0.0;
+            int i = 0;
+            for (Player p : team.squad) {
+                while (i < 11) {
+                    i++;
+                    predictedScore += p.averageRuns;
+                }
+            }
+            mapTeamToPredictedScore.put(team, predictedScore);
+        }
+
+        Map<Team, Double> sorted = mapTeamToPredictedScore
+                .entrySet()
+                .stream()
+                .sorted(comparingByValue())
+                .collect(
+                        toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2,
+                                LinkedHashMap::new));
+
+        int i = 0;
+        for (Map.Entry<Team, Double> entry : sorted.entrySet()) {
+            if (i == 6) {
+                System.out.println("Runner: " + entry.getKey().teamName + " Predicted Score: " + entry.getValue());
+            }
+            if (i == 7) {
+                System.out.println("Winner: " + entry.getKey().teamName + " Predicted Score: " + entry.getValue());
+            } else {
+                i++;
+            }
```

Comment:
The displaying of winner and runner can be moved to a separate function 
The logic also needs to be better